support/mender-inventory-geo

```diff
@@ -35,27 +30,33 @@ err() {
 # This value is in seconds
 TIMEOUT=10
 
-# find the ip address
-ip=`wget --timeout=${TIMEOUT} -qO /dev/stdout --header "Content-Type: text/plain;" \
-     --header 'Host: ifconfig.io' --header 'User-Agent: curl/7.67.0' \
-     --header 'Accept: */*' ifconfig.io | tail -1`
-
-[ -n "${ip}" ] || err 2 "Unable to get the IP address from ifconfig.io"
-
-# get the information, see https://www.ipvigilante.com/api-developer-docs/
-geo=`wget --timeout=${TIMEOUT} -qO /dev/stdout https://ipvigilante.com/csv/${ip} 2>/dev/null`
-
-[ -n "${geo}" ] || err 3 "Unable to get the geolocalization data from ipvigilante.com"
+ipinfo=$(wget --timeout=${TIMEOUT} -q -O /dev/stdout \
+              --header 'Accept: application/json' \
+              https://ipinfo.io)
+if [ $? != 0 ] || [ -z "${ipinfo}" ]; then
+    err 2 "Unable to get IP info from ipinfo.io"
+fi
+# Convert JSON (Object{string -> string}) to key=value format.
+ipinfo=$(echo "${ipinfo}" | \
+             tr -d "\n" | \
+             sed 's/^\s*{\(.*\)\s*"\s*}\s*$/\1/' | \
+             sed 's/\([^\\]\)",/\1\n/g' | \
+             sed 's/^\s*"//' | \
+             sed 's/\([^\\]\)\s*"\s*:\s*"/\1=/'
+      )
 
 # the names of the attributes holding the localization data can be configured here
 ATTR_NAME_IP="geo-ip"
-ATTR_NAME_CONTINENT="geo-continent"
-ATTR_NAME_COUNTRY="geo-country"
 ATTR_NAME_CITY="geo-city"
-
-echo "${geo}" | awk -v ip="${ATTR_NAME_IP}" -v continent="${ATTR_NAME_CONTINENT}" \
-    -v country="${ATTR_NAME_COUNTRY}" -v city="${ATTR_NAME_CITY}" -F',' \
-    ' length($NF) == 0 { exit(4) }
-      { printf("%s=%s\n%s=%s\n%s=%s\n%s=%s\n",ip,$1,continent,$2,country,$3,city,$6) }
-    '
-
+ATTR_NAME_COUNTRY="geo-country"
+ATTR_NAME_TIMEZONE="geo-timezone"
+
+echo "${ipinfo}" | awk -F'=' -v attr_ip="${ATTR_NAME_IP}" \
+                       -v attr_city="${ATTR_NAME_CITY}" \
+                       -v attr_country="${ATTR_NAME_COUNTRY}" \
+                       -v attr_zone="${ATTR_NAME_TIMEZONE}" '
+$1 ~ /^ip/{printf "%s=%s\n", attr_ip, $2};
+$1 ~ /^city/{printf "%s=%s\n", attr_city, $2};
```

Comment:
Suggestion:
A little bit of air between the `pattern` and the `expression` would be nice actually, so maybe something like:
```
echo "${ipinfo}" | awk -F'=' -v attr_ip="${ATTR_NAME_IP}" \
                       -v attr_city="${ATTR_NAME_CITY}" \
                       -v attr_country="${ATTR_NAME_COUNTRY}" \
                       -v attr_zone="${ATTR_NAME_TIMEZONE}" '

$1 ~ /^ip/       {printf "%s=%s\n", attr_ip, $2}
$1 ~ /^city/     {printf "%s=%s\n", attr_city, $2}
$1 ~ /^country/  {printf "%s=%s\n", attr_country, $2}
$1 ~ /^timezone/ {printf "%s=%s\n", attr_zone, $2}
'
```